patches/api/0197-Spawn-Reason-API.patch

```diff
@@ -5,50 +5,150 @@ Subject: [PATCH] Spawn Reason API
++    }
++
++    /**
++     * Creates an arrow entity of the given class at the given {@link Location}
++     *
++     * @param <T> type of arrow to spawn
++     * @param location Location to spawn the arrow
++     * @param direction Direction to shoot the arrow in
++     * @param speed Speed of the arrow. A recommend speed is 0.6
++     * @param spread Spread of the arrow. A recommend spread is 12
++     * @param clazz the Entity class for the arrow
++     * {@link org.bukkit.entity.SpectralArrow SpectralArrow},{@link org.bukkit.entity.Arrow Arrow},{@link org.bukkit.entity.TippedArrow TippedArrow}
++     * @param reason the spawn reason
++     * @return Arrow entity spawned as a result of this method
++     */
++    @NotNull
++    <T extends AbstractArrow> T spawnArrow(@NotNull Location location, @NotNull Vector direction, float speed, float spread, @NotNull Class<T> clazz, @NotNull org.bukkit.event.entity.CreatureSpawnEvent.SpawnReason reason);
++    // Paper end
+ 
+     /**
+      * Creates a tree at the given {@link Location}
+@@ -749,16 +769,44 @@ public interface World extends RegionAccessor, WorldInfo, PluginMessageRecipient
+      * @return The lightning entity.
+      */
+     @NotNull
+-    public LightningStrike strikeLightning(@NotNull Location loc);
++    // Paper start
++    default LightningStrike strikeLightning(@NotNull Location loc) {
++        return strikeLightning(loc, org.bukkit.event.weather.LightningStrikeEvent.Cause.CUSTOM);
++    }
++
++    /**
++     * Strikes lightning at the given {@link Location}
++     *
++     * @param loc The location to strike lightning
++     * @param cause The cause of this lightning
++     * @return The lightning entity.
++     */
++    @NotNull
++    LightningStrike strikeLightning(@NotNull Location loc, @NotNull org.bukkit.event.weather.LightningStrikeEvent.Cause cause);
++    // Paper end
++
++    /**
++     * Strikes lightning at the given {@link Location} without doing damage
++     *
++     * @param loc The location to strike lightning
++     * @return The lightning entity.
++     */
++    @NotNull
++    // Paper start
++    default LightningStrike strikeLightningEffect(@NotNull Location loc) {
++        return strikeLightningEffect(loc, org.bukkit.event.weather.LightningStrikeEvent.Cause.CUSTOM);
++    }
+ 
+     /**
+      * Strikes lightning at the given {@link Location} without doing damage
+      *
+      * @param loc The location to strike lightning
++     * @param cause The cause of this lightning
+      * @return The lightning entity.
+      */
+     @NotNull
+-    public LightningStrike strikeLightningEffect(@NotNull Location loc);
++    LightningStrike strikeLightningEffect(@NotNull Location loc, @NotNull org.bukkit.event.weather.LightningStrikeEvent.Cause cause);
++    // Paper end
```

Comment:
End the paper comment block up top with a single inline comment on the last line.

Reply:
Idk why but the diff has changed during the rebasing.